### ISSUE TITLE
REST API support

### ISSUE THREAD

**Issue by mumoshu** [radu-matei/kube-toolkit#28]
It would be nice include a library or a stub of a REST API server in kube-toolkit generated project, so that we could build a nifty Web UI which relies on the REST API.
Imagine something like Brigade + Kashti :)

For reference, it would be nice to have a stub implementation of a REST API server like [Brigade's](https://github.com/Azure/brigade/blob/master/brigade-api/cmd/brigade-api/main.go).
I'm not yet sure which part of the stub could be extracted into a library of kube-toolkit(Nothing?).

**Comment by radu-matei** [comment]
Have you seen this?

![gif](https://raw.githubusercontent.com/radu-matei/kube-toolkit/master/docs/img/proxy.gif)

There actually is a dashboard in /gateway/web, using [gRPC-gateway](https://github.com/grpc-ecosystem/grpc-gateway) - essentially, add this to your gRCP method definition:

```
	rpc GetVersion(google.protobuf.Empty) returns (Version){
		option (google.api.http) = {
            get: "/api/version"
          };
	}
```

And when building you automatically get a Swagger definition, together with a TypeScript client ready to use in the Angular dashboard app.

Does this answer your question?

(There should be a very clear doc explaining how to extend and add a web component, but that's another story...)

**Comment by mumoshu** [comment]
@radu-matei Awesome! It seems to be exactly what I was looking for. Great job 👍

**Comment by mumoshu** [comment]
Thanks for the answer. Closing this as resolved, opened a dedicated issue #30 to track documentation improvements.
